### DISCUSSION TITLE
How to solve fetch failed error?

### DISCUSSION THREAD

**Discussion by realgump** [danny-avila/LibreChat#106] [Troubleshooting] [answered]
![image](https://user-images.githubusercontent.com/37771769/226623041-27bb8e28-53be-4658-80b9-811d30976306.png)

**Comment by realgump** [comment]
And there are logs of server:
```
chatgpt-clone-api-1     | model: chatgpt oldConvoId: null
chatgpt-clone-api-1     | conversationId after old: a7307e5d-a432-456c-bd1f-e9df98e39ae6
chatgpt-clone-api-1     | ask log {
chatgpt-clone-api-1     |   model: 'chatgpt',
chatgpt-clone-api-1     |   messageId: '77f48ea8-60c5-40b4-b4cd-c620f7257f31',
chatgpt-clone-api-1     |   sender: 'User',
chatgpt-clone-api-1     |   text: 'Explain quantum computing in simple terms',
chatgpt-clone-api-1     |   parentMessageId: '00000000-0000-0000-0000-000000000000',
chatgpt-clone-api-1     |   conversationId: 'a7307e5d-a432-456c-bd1f-e9df98e39ae6',
chatgpt-clone-api-1     |   isCreatedByUser: true,
chatgpt-clone-api-1     |   current: true,
chatgpt-clone-api-1     |   chatGptLabel: null,
chatgpt-clone-api-1     |   promptPrefix: null
chatgpt-clone-api-1     | }
chatgpt-clone-api-1     | TypeError: fetch failed
chatgpt-clone-api-1     |     at Object.fetch (node:internal/deps/undici/undici:11457:11)
chatgpt-clone-api-1     |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
chatgpt-clone-api-1     |     at async create (file:///api/node_modules/@waylaidwanderer/fetch-event-source/lib/esm/fetch.js:52:34) {
chatgpt-clone-api-1     |   cause: Error: connect ECONNREFUSED 127.0.0.1:7896
chatgpt-clone-api-1     |       at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1532:16) {
chatgpt-clone-api-1     |     errno: -111,
chatgpt-clone-api-1     |     code: 'ECONNREFUSED',
chatgpt-clone-api-1     |     syscall: 'connect',
chatgpt-clone-api-1     |     address: '127.0.0.1',
chatgpt-clone-api-1     |     port: 7896
chatgpt-clone-api-1     |   }
chatgpt-clone-api-1     | }
```

**Reply by ladypiapia** [reply]
Have you solved the problem？ I also encountered the same problem, I know it is a proxy problem, but do not know how to solve it.Should I buy a foreign server to build a proxy myself? I also use VPN on a daily basis, but I don't know how to use proxies in servers or such code

**Reply by danny-avila** [reply]
> Have you solved the problem？ I also encountered the same problem, I know it is a proxy problem, but do not know how to solve it.Should I buy a foreign server to build a proxy myself? I also use VPN on a daily basis, but I don't know how to use proxies in servers or such code

Sometimes VPNs also include proxies in their package. See if your provider has one, or if you can buy from them if not

**Reply by realgump** [reply]
> Have you solved the problem？ I also encountered the same problem, I know it is a proxy problem, but do not know how to solve it.Should I buy a foreign server to build a proxy myself? I also use VPN on a daily basis, but I don't know how to use proxies in servers or such code

I don't think my issue is caused by an IP block, as I am able to successfully receive a response using CURL. In fact, I used a Docker container to run the code. However, it was later discovered that the Docker container had an incorrect network setting,  since it was unable to access any websites.

**Reply by danny-avila** [reply]
Sorry about the inconvenience, yes I’ve also had issues with cross-container networks. I’ve now specified more explicitly the setting needed for that in the docker compose

**Comment by danny-avila** [comment, marked as answer]
Your IP is blocking you https://github.com/waylaidwanderer/node-chatgpt-api/issues/177#issuecomment-1464681067

You will need a proxy which you can set in your env file
